sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -234,7 +235,7 @@ case class DropTableCommand(
       }
     }
 
-    if (isTempView || catalog.tableExists(tableName)) {
+    if (isTempView || isTableExist) {
```

Comment:
Hm, @wangyum, when `isTempView` is `true`, `catalog.tableExists(tableName)` won't be called by short-circuiting. So, this isn't a clean improvement but it has trade-off. Can you clarify this trade-off as well?